spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -676,10 +831,85 @@ def setup_and_check(self, data, title=''):
         self.setWindowTitle(title)
         self.resize(600, 500)
 
+        self.hscroll = QScrollBar(Qt.Horizontal)
+        self.vscroll = QScrollBar(Qt.Vertical)
+
+        # Create the view for the level
+        self.table_level = QTableView()
+        self.table_level.setEditTriggers(QTableWidget.NoEditTriggers)
+        self.table_level.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.table_level.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.table_level.setFrameStyle(QFrame.Plain)
+        self.table_level.horizontalHeader().sectionResized.connect(
+                                                        self._index_resized)
+        self.table_level.verticalHeader().sectionResized.connect(
+                                                        self._header_resized)
+        self.table_level.setItemDelegate(QItemDelegate())
+        self.layout.addWidget(self.table_level, 0, 0)
+        self.table_level.setContentsMargins(0, 0, 0, 0)
+        self.table_level.horizontalHeader().sectionClicked.connect(
+                                                            self.sortByIndex)

```

Comment:
@dalthviz, please move almost all these lines to a new method called `create_table_level`, so that this looks like

```python
self.table_level = self.create_table_level()
self.layout.addWidget(self.table_level, 0, 0)

def create_table_level(self):
    self.table_level = QTableView()
    self.table_level.setEditTriggers(QTableWidget.NoEditTriggers)
    self.table_level.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
    self.table_level.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
    self.table_level.setFrameStyle(QFrame.Plain)
    self.table_level.horizontalHeader().sectionResized.connect(
                                                    self._index_resized)
    self.table_level.verticalHeader().sectionResized.connect(
                                                    self._header_resized)
    self.table_level.setItemDelegate(QItemDelegate())
    self.table_level.setContentsMargins(0, 0, 0, 0)
    self.table_level.horizontalHeader().sectionClicked.connect(
                                                    self.sortByIndex)
```